spec/providers/atmosphere/local_pdp_spec.rb

```diff
@@ -0,0 +1,88 @@
+require 'rails_helper'
+
+describe Atmosphere::LocalPdp do
+  context '#can_manage?' do
+    let(:user) { build(:user, id: 1) }
+    subject { Atmosphere::LocalPdp.new(user) }
+
+    it 'allows to manage owned appliance types' do
+      at = build(:appliance_type, user_id: user.id)
+
+      expect(subject.can_manage?(at)).to be_truthy
+    end
+
+    it 'allows to manage appliance types where management permissions have been granted' do
+      at = build(:appliance_type)
+      create(:user_appliance_type, user: user, appliance_type: at, role: 'manager')
+
+      expect(subject.can_manage?(at)).to be_truthy
+    end
+
+    it 'does not allow to manage unowned appliance types' do
+      at = build(:appliance_type)
```

Comment:
Yes, but there's an exception for services flagged as owned by the current user - see line 36 in `local_pdp.rb`:

<pre>
      obj.user_id == current_user.id || uat.present?
</pre>

Since `at` is passed directly to the PDP method, this condition will succeed without running a DB query.